src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp

```diff
@@ -97,36 +97,85 @@ double get_temporal_id_value(const T& id) {
 CREATE_IS_CALLABLE(apply)
 CREATE_IS_CALLABLE_V(apply)
 
-// apply_callback accomplishes the overload for the
-// two signatures of callback functions.
+template <typename Callbacks, typename Box, typename Cache, typename TemporalId>
+struct AllSimpleCallbacks;
+
+template <typename... Callbacks, typename Box, typename Cache,
+          typename TemporalId>
+struct AllSimpleCallbacks<tmpl::list<Callbacks...>, Box, Cache, TemporalId>
+    : public std::integral_constant<
+          bool, tmpl2::flat_all_v<is_apply_callable_v<Callbacks, Box, Cache,
+                                                      TemporalId>...>> {};
```

Comment:
[optional] You can just inherit from `tmpl2::flat_all<...>` and skip the `type -> bool -> type` conversion, since that already inherits from an `integral_constant`.  

As another alternative, you could use a fold expression.  I don't know if anyone's compared performance between those and `flat_all`, but these are small lists so it shouldn't matter.

And yet another alternative:  Is
```c++
AllSimpleCallbacks<callbacks, decltype(*box), decltype(*cache),
                   decltype(temporal_id)>
```
sufficiently clearer than
```c++
tmpl::all<callbacks,
          is_apply_callable<tmpl::_1, tmpl::pin<decltype(*box)>,
                            tmpl::pin<decltype(*cache)>,
                            tmpl::pin<decltype(temporal_id)>>
```
to be worth the extra struct?